### PR TITLE
Go API for EVM precompiles

### PR DESCRIPTION
**Pull request by Vindaar**
This continues from #383 by porting over the EVM precompiles C API into Go. All test cases are ported from Nim and should all pass.

### CODE REVIEW

File: constantine-go/constantine.go [hunk 1]
```diff
@@ -265,4 +265,585 @@ func (ctx EthKzgContext) VerifyBlobKzgProofBatchParallel(tp Threadpool, blobs []
 		return false, err
 	}
 	return true, nil
-}
\ No newline at end of file
+}
+
+// Ethereum BLS signatures
+// -----------------------------------------------------
```
Comment (mratsim) [inline constantine-go/constantine.go:271]: I think this needs a rebase on master as this should already be there from #383 

See https://github.com/mratsim/constantine/pull/383/files#diff-66f2a828703913aa04c16d50c2e86f4f3ffd7551cccee3a74a6ddafc00acad75R271

**Review by mratsim — COMMENTED** [review]
I think this is OK but large interference from #383 that needs a rebase.

**Comment by Vindaar** [comment]
Woops, guess I screwed up the rebase I did. Will investigate.

File: constantine-go/constantine_test.go [hunk 2]
```diff
@@ -1379,3 +1379,232 @@ func TestBatchVerify(t *testing.T) {
...
+	multiexp_G2_bls_tests			= filepath.Join(testDirEvm, "eip-2537/multiexp_G2_bls.json")
+	fail_multiexp_G2_bls_tests		= filepath.Join(testDirEvm, "eip-2537/fail-multiexp_G2_bls.json")
+
+	pairing_check_bls_tests			= filepath.Join(testDirEvm, "eip-2537/pairing_check_bls.json")
+	fail_pairing_check_bls_tests	= filepath.Join(testDirEvm, "eip-2537/fail-pairing_check_bls.json")
+
+	map_fp_to_G1_bls_tests			= filepath.Join(testDirEvm, "eip-2537/map_fp_to_G1_bls.json")
+	fail_map_fp_to_G1_bls_tests		= filepath.Join(testDirEvm, "eip-2537/fail-map_fp_to_G1_bls.json")
+	map_fp2_to_G2_bls_tests			= filepath.Join(testDirEvm, "eip-2537/map_fp2_to_G2_bls.json")
+	fail_map_fp2_to_G2_bls_tests	= filepath.Join(testDirEvm, "eip-2537/fail-map_fp2_to_G2_bls.json")
```
Comment (Vindaar) [inline constantine-go/constantine_test.go:1450]: Tabs are annoying. With 4 space width equivalent tabs the lines are aligned on the `=`...

Comment (mratsim) [reply constantine-go/constantine_test.go:1450]: I agree, but gofmt dictatorship

**Review by mratsim — APPROVED** [review]

**Pull request merged**